ui/src/app/pages/admin/admin-user.tsx

```diff
@@ -136,32 +145,134 @@ const AdminUser = withUrlParams()(class extends React.Component<Props, State> {
     const {gsuiteDomain} = serverConfigStore.getValue();
     try {
       const profile = await profileApi().getUserByUsername(this.props.urlParams.usernameWithoutGsuiteDomain + '@' + gsuiteDomain);
-      this.setState({profile: profile});
+      this.setState({oldProfile: profile, updatedProfile: profile});
     } catch (error) {
       this.setState({profileLoadingError: 'Could not find user - please check spelling of username and try again'});
     }
   }
 
   async getInstitutions() {
     try {
-      const institutionsResponse = await institutionApi().getInstitutions();
-      const options = fp.map(
-        institution => {
-          return {
-            'label': institution.displayName ? institution.displayName : institution.shortName,
-            'value': {displayName: institution.displayName, shortName: institution.shortName}
-          };
-        },
-        institutionsResponse.institutions
-      );
-      this.setState({verifiedInstitutionOptions: options});
+      const institutionsResponse = await institutionApi().getPublicInstitutionDetails();
+      const institutions = institutionsResponse.institutions;
+      this.setState({
+        verifiedInstitutionOptions: fp.sortBy( institution => institution.displayName, institutions),
+        verifiedInstitutionsByShortname: institutions.reduce(
+          (accumulator, institution) => accumulator.set(institution.shortName, institution),
+          new Map<String, PublicInstitutionDetails>()
+        )
+      });
     } catch (error) {
       this.setState({institutionsLoadingError: 'Could not get list of verified institutions - please try again later'});
     }
   }
 
+  getRoleOptionsForProfile() {
+    const {updatedProfile: {verifiedInstitutionalAffiliation}, verifiedInstitutionOptions} = this.state;
+    const institutionShortName = verifiedInstitutionalAffiliation ? verifiedInstitutionalAffiliation.institutionShortName : '';
+    return getRoleOptions(verifiedInstitutionOptions, institutionShortName);
+  }
+
+  getInstitutionDropdownOptions() {
+    const {verifiedInstitutionOptions} = this.state;
+    return fp.map(
+      institution => {
+        return {label: institution.displayName, value: institution.shortName};
+      },
+      verifiedInstitutionOptions
+    );
+  }
+
+  isSaveDisabled(errors) {
+    const {oldProfile, updatedProfile} = this.state;
+    return fp.isEqual(oldProfile, updatedProfile) || errors;
+  }
+
+  setVerifiedInstitutionOnProfile(institutionShortName: string) {
+    const {verifiedInstitutionsByShortname} = this.state;
```

Comment:
Consider trying to perform the state change in one operation with flow:
```suggestion
    this.setState(fp.flow(
        fp.set(['updatedProfile', 'verifiedInstitutionalAffiliation', 'institutionShortName'],institutionShortName),
        fp.set(['updatedProfile', 'verifiedInstitutionalAffiliation', 'institutionDisplayName'], fp.find(['shortName', institutionShortName], sortedInstitutions).displayName),
        fp.set(['updatedProfile', 'verifiedInstitutionalAffiliation', 'institutionRoleEnum'], undefined),
        fp.set(['updatedProfile', 'verifiedInstitutionalAffiliation', 'institutionalRoleOtherText'], undefined)
    ))
```

Reply:
This works, but not if I combine it with other state changes for some reason